source/org.jetbrains.kotlinx/kotlinx-serialization-protobuf-jvm/Additions/KotlinX.Serialization.Protobuf.ProtoNumberImpl.cs

```diff
@@ -0,0 +1,16 @@
+namespace KotlinX.Serialization.Protobuf 
+{
+
+	public sealed partial class ProtoNumberImpl
```

Comment:
I have two concerns with these `*Impl` types:

 1. Do they need to exist *at all*?  I'm reluctant to expose types/packages with "internal" or "impl[ementation]" in their name.

    Additionally, these `Impl` nested types aren't even listed in the documentation! Consider the [`ProtoNumber`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-protobuf/kotlinx.serialization.protobuf/-proto-number/) docs, which makes no mention of nested types or `Impl`.

 2. Assuming that we *do* want to expose this type, this feels like the wrong "scope".

Consider [kotlinx-serialization-protobuf-jvm-1.7.0.jar](https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-serialization-protobuf-jvm/1.7.0/kotlinx-serialization-protobuf-jvm-1.7.0.jar):

```sh
% javap -cp kotlinx-serialization-protobuf-jvm-1.7.0.jar 'kotlinx/serialization/protobuf/ProtoNumber'     
Compiled from "ProtoTypes.kt"
public interface kotlinx.serialization.protobuf.ProtoNumber extends java.lang.annotation.Annotation {
  public abstract int number();
}

% javap -cp kotlinx-serialization-protobuf-jvm-1.7.0.jar 'kotlinx/serialization/protobuf/ProtoNumber$Impl'      
Compiled from "ProtoTypes.kt"
public final class kotlinx.serialization.protobuf.ProtoNumber$Impl implements kotlinx.serialization.protobuf.ProtoNumber {
  public kotlinx.serialization.protobuf.ProtoNumber$Impl(int);
  public final int number();
}
```

which implies a binding of:

```csharp
public partial interface IProtoNumber : Java.Lang.Annotation.IAnnotation {
    public partial class Impl : IProtoNumber {
    }
}
```

i.e. `IProtoNumber.Impl`, *not* `ProtoNumberImpl`.

Reply:
removed